src/main/thrift/parquet.thrift

```diff
@@ -639,24 +639,47 @@ struct OffsetIndex {
 struct ColumnIndex {
 /**
  * A list of bools to determine the validity of the corresponding min and max
- * values. If this is true, the corresponding min and max values in the next
- * two lists are non-null.
+ * values. If true, the page contains only null values so the corresponding
+ * entries in min_values and max_values should be ignored.
  */
   1: required list<bool> null_pages
 
 /**
- * A list containing the lower bounds for the values of each page. This may be
- * the actual minimum value found on a page, but can also be a (more compact)
- * value that does not exist on that page. For ordered columns, these values
- * must be lower and upper bounds for consecutive pages respectively.
+ * Two lists containing lower and upper bounds for the values of each page.
+ * These may be the actual minimum and maximum values found on a page, but can
+ * also be (more compact) values that does not exist on a page.
+ *
+ * For ordered columns, max_values can be ommitted. In this case min_values
+ * must contain values that are both lower and upper bounds for consecutive
+ * pages respectively.
+ *
+ * Example: Consider the following values in pages delimited by |:
+ * | 1 1 2 3 | 5 8 13 21 | 34 55 89 144 |                       ^
+ * ^         ^           ^              ^
+ * Examples for valid min_values are [1 5 34], [-2 4 22], or [1 3 21].
+ *
+ * Note how the maximum value of a page is also a valid minimum boundary for
+ * the next page. In general, readers must assume that a minimum value also
+ * occurs in the previous page.
+ *
+ * For columns in descending order, min_values still stores lower bounds, i.e.
+ * values that come first in the ordering.
+ *
+ * Consider the following values:
+ * | 37 31 29 23 | 19 17 13 11 | 7 5 3 2 |
+ * ^             ^             ^         ^
+ * Examples for valid lists of minimum values are [23 11 2] or [ 20 8 0 ].
```

Comment:
I think I disagree? Min should respect the sort order, so a descending order produces `23 < 11 => true`. Valid mins for this example are `[37, 19, 7]` or `[38, 20, 8]` or `[37, 23, 11]`

Reply:
On second thought, I don't think it is a good idea to do this. The mins should always be the min defined by the type's order. Descending order shouldn't change that, but it does make the logic more confusing when trying to use it as both min and max.

When mins are min and max, descending order changes the page for which the min of P_i min is a max. For asecending order:

<img src="https://latex.codecogs.com/gif.latex?min(P_i)&space;\approx&space;max(P_{i-1})" title="min(P_i) \approx max(P_{i-1})" />

For descending:

<img src="https://latex.codecogs.com/gif.latex?min(P_i)&space;\approx&space;max(P_{i&plus;1})" title="min(P_i) \approx max(P_{i+1})" />

That's annoying. Maybe we should just always require `max_values` as Julien suggests. That makes it easy to produce smaller bounds anyway by simplifying the constraints for truncation.

Reply:
For sorted cases we can do binary search, so even if both lists are required, I think it would be helpful to indicate whether they are ordered.

I don't see Julien's comment on that. Can you point me to it?

Reply:
The asymmetry is indeed annoying, but I don't see a nice way of getting around it.

Reply:
I'm ok with making max_values required, too. However, if min_values is ordered and partitions the values, then one can use binary search to find matching pages. To enable that, we would need some boolean that writers can set to indicate this property.

@julienledem I couldn't reply to your comment above for some Github reason. :(